src/element-stub.js

```diff
@@ -15,14 +15,31 @@
  */
 
 import {BaseElement} from './base-element';
+import {getMode} from '../src/mode'
 
 
 /** @type {!Array} */
 export const stubbedElements = [];
 
+let ampAdScriptInserted = false;
+
 export class ElementStub extends BaseElement {
   constructor(element) {
     super(element);
+    if((element.tagName.toLowerCase() == "amp-ad"
+      || element.tagName.toLowerCase() == "amp-embed") 
+      && (!ampAdScriptInserted)) {
+      var ampAdScript = this.getWin().document.createElement('script');
+      ampAdScript.async = true;
+      ampAdScript.setAttribute('custom-element', 'amp-ad');
```

Comment:
Please add another `data-` attribute, so it is visible in the DOM that it went through this code path.